[2] docs: previewer explanation (#3976)
inveniosoftware/invenio: docs/managing-files/previewer.rst

The available previewers can be configured in ``PREVIEWER_PREFERENCE`` in
``config.py`` which is a list of previewers. The order in which they appear in
the list will be the order they are used when finding a suitable previewer.

Invenio-Previewer supports previewing images out-of-the-box however it's very
basic. If you need more advanced image support, like thumbnails, please see
:ref:`preview-with-iiif`.

For a full example of how to use Invenio-Previewer, please see the
`usage <https://invenio-previewer.readthedocs.io/en/latest/usage.html>` guide.

.. note::

    **Are you trying to preview a file format which is not supported?** Please
    see the section about `developing your own previewer <https://invenio-previewer.readthedocs.io/en/latest/usage.html#custom-previewer>`_ for Invenio-Previewer.

.. _preview-with-iiif:

Preview images using IIIF
-------------------------
Invenio-IIIF implements the `IIIF API <https://iiif.io/>`_ which is a standard
for describing and delivering images over the web. `Invenio-IIIF <https://invenio-iiif.readthedocs.io/en/latest/>`_
allows you to generate thumbnails, resize and zoom images as well as previewing
images using Invenio-Previewer.

> For Invenio-IIIF to work the ``iiif_image`` previewer needs to be activated

Reviewer: ```suggestion
To activate the IIIF Image previewer you need to add ``iiif_image`` into the list of supported previewers ``PREVIEWER_PREFERENCE``, e.g.:
```

which is done by updating ``PREVIEWER_PREFERENCE`` mentioned earlier, e.g.:

.. code:: python

    PREVIEWER_PREFERENCE = [
        'iiif_image',
        'json_prismjs',
        'xml_prismjs',
        'pdfjs',
        'zip',
    ]

The IIIF API is available by default on ``/api/iiif`` and to generate
valid image URLs, the utilify function `invenio_iiif.utils.ui_iiif_image_url <https://invenio-iiif.readthedocs.io/en/latest/api.html#invenio_iiif.utils.ui_iiif_image_url>`_
can be used. Given that you have an image object ``img_obj`` belonging to a bucket
``bucket`` (see :ref:`managing-files`), an image URL can be generated for that

> image like this:

Reviewer: ```suggestion
image as in the example below:
```

.. code:: python

    from invenio_iiif.utils import ui_iiif_image_url
    image_url = ui_iiif_image_url(
          obj=img_obj, version='v2', region='full', size='full', rotation=0,
          quality='default', image_format='png')

The generated image URL will look similar to this:

``/iiif/<bucket_id>:<version_id>:img.png/full/full/0/default.png``

> For a more complete guide on how to use Invenio-IIIF see the

Reviewer: ```suggestion
For a more complete guide on how to use Invenio-IIIF and detailed documentation, see the
```

..
    This file is part of Invenio.
    Copyright (C) 2019 CERN.

    Invenio is free software; you can redistribute it and/or modify it
    under the terms of the MIT License; see LICENSE file for more details.

Previewing files
================
Invenio has support for previewing many of the most popular file formats
including PDF, ZIP, Markdown, images and Jupyter Notebooks. In this section we
will discuss how previewing files work in Invenio and which options are
available.

`Invenio-Previewer <https://invenio-previewer.readthedocs.io/en/latest/>`_
allows you to register previewers based on the file extension. To preview

> files, ensure that a preview endpoint have been configured in your

Reviewer 2: I might be wrong, but I think it's more common to use`has been` with the third person?

``config.py``, like this:

.. code:: python

    RECORDS_UI_ENDPOINTS=dict(
        recid_previewer=dict(
            pid_type='recid',
            route='/records/<pid_value>/preview/<filename>',
            view_imp='invenio_previewer.views:preview',
            record_class='invenio_records_files.api:Record',
        ),
    )

> Upload a file (see :ref:`managing-files`) of a type supported by the previewer,

Reviewer 2: Once, there is a full `managing-files` section, we should update the reference to point to the exact file upload step.
